addons/hr_expense/models/hr_expense.py

```diff
@@ -1176,8 +1176,14 @@ def _prepare_payment_vals(self):
             amount = sum(self.expense_line_ids.mapped('total_amount'))
         move_lines = []
         for expense in self.expense_line_ids:
-            tax_data = self.env['account.tax']._compute_taxes([expense._convert_to_tax_base_line_dict(price_unit=expense.total_amount, currency=expense.currency_id)])
-            rate = abs(expense.total_amount / expense.total_amount_company)
+            if currency == expense.company_currency_id:
+                expense_amount = expense.total_amount_company
+            else:
+                expense_amount = expense.total_amount
```

Comment:
```suggestion
            expense_amount = expense.total_amount_company if self.is_multiple_currency else expense.total_amount
```
I think this should work, and is a bit cleaner IMO.